packages/idx/src/property/search.component.html

```diff
@@ -15,279 +15,346 @@ <h2 data-ng-show="widgetName" data-ng-bind="widgetName"></h2>
+            <!-- NOTE: some of these fields are arrays -->
+            <md-icon class="search-icon"
+                     data-ng-click="searchProperties()"
+                     data-ng-if="!(options.query.where.City || options.query.where.CountyOrParish.length || options.query.where.MLSAreaMajor.length || options.query.where.PostalCode.length)"
+                     md-svg-src="{{ 'sitetheorycore/images/icons/actionButtons/search.svg' | assetPath }}"></md-icon>
+            <md-input-container
+                    class="md-block minimal"
+                    data-ng-if="!(options.query.where.City || options.query.where.CountyOrParish.length || options.query.where.MLSAreaMajor.length || options.query.where.PostalCode.length)">
+                <label>City, Neighborhood, Zip</label>
+
+                <!-- TODO: the ng-keyup should not be setting: && options.query.where.Location (seems useless???)  -->
+                <input id="filterLocation"
+                       data-ng-model="options.query.where.Location"
+                       data-ng-blur="searchProperties()"
+                       type="text"
+                       length="22"
+                       maxlength="250"
+                       data-ng-keyup="$event.keyCode == 13 && searchProperties()"
+                       autocomplete="off"
+                >
             </md-input-container>
+
+            <!-- TODO: for now we are not setting the location to the combined presets because we can't get the focus working. But if we do, the code will need to be fixed because CountyOrParish, MLSAreaMajor and PostalCode are arrays not strings: data-ng-click="options.query.where.Location = _.trim(options.query.where.City + ' ' + options.query.where.CountyOrParish + ' ' + options.query.where.MLSAreaMajor + ' ' + options.query.where.PostalCode); options.query.where.City = null; options.query.where.CountyOrParish = null; options.query.where.MLSAreaMajor = null; options.query.where.PostalCode = null;"
+            -->
+
+            <!-- TODO: make this gray in the CSS -->
+            <div class="preset-location"
+                 data-ng-if="(options.query.where.City.length || options.query.where.CountyOrParish.length || options.query.where.MLSAreaMajor.length || options.query.where.PostalCode.length)">
+                <span data-ng-if="options.query.where.City.length" ng-bind="options.query.where.City"></span>
+                <!-- NOTE: these are actually arrays so we union them then join split with a comma -->
+                <span data-ng-if="options.query.where.CountyOrParish.length || options.query.where.MLSAreaMajor.length || options.query.where.PostalCode" ng-bind="(options.query.where.City.length && (options.query.where.CountyOrParish.length || options.query.where.MLSAreaMajor.length || options.query.where.PostalCode.length) ? ', ' : '') + _.join(_.union(options.query.where.CountyOrParish, options.query.where.MLSAreaMajor, options.query.where.PostalCode), ', ')"></span>
```

Comment:
angular has a `ng-list` option where it'll convert the array into a comma'ed string